arcade/arcade/core/schema.py

```diff
@@ -262,14 +262,20 @@ class ToolCallRequest(BaseModel):
     """The context for the tool invocation."""
 
 
-class ToolCallWarning(BaseModel):
-    """A warning that occurred during the tool invocation."""
+class ToolCallLog(BaseModel):
+    """A log that occurred during the tool invocation."""
 
     message: str
     """The user-facing warning message."""
 
-    warning_type: Literal["deprecation"]
-    """The type of warning that occurred."""
+    level: Literal[
+        "debug",
+        "info",
+        "warning",
+        "error",
+        "deprecation",
```

Comment:
Why should `deprecation` be its own level, rather than being a `warning`?

Reply:
I chose this way because deprecation felt like it is a special type of warning given that it is a warning about the tool itself, and not a warning that occurred during the execution of the tool.

I see your concern since typically the levels are debug, info, warning, error.

If we don't think it warrants its own level, then changing this would be very easy.

Reply:
Ah, I see what you mean by it being a special type. It could be useful to handle that type differently downstream.

To me, that feels like something that could go into a `type` field, rather than overloading `level` (since loglevels are a pretty well-defined thing in most code)

Reply:
yep, an optional type field was the alternative. I'll see what I can come up with

Reply:
I added `subtype` field to `ToolCallLog`.